probe-rs/src/architecture/xtensa/mod.rs

```diff
@@ -101,6 +105,54 @@ impl<'probe> Xtensa<'probe> {
 
         Ok(())
     }
+
+    /// Check if the current breakpoint is a semihosting call
+    // OpenOCD implementation: https://github.com/espressif/openocd-esp32/blob/93dd01511fd13d4a9fb322cd9b600c337becef9e/src/target/espressif/esp_xtensa_semihosting.c#L42-L103
+    fn check_for_semihosting(
+        old_reason: HaltReason,
+        core: &mut dyn CoreInterface,
+    ) -> Result<HaltReason, Error> {
+        let mut reason = old_reason;
+        let pc: u32 = core.read_core_reg(core.program_counter().id)?.try_into()?;
+
+        let mut actual_instructions = [0u32; 1];
+        core.read_32((pc) as u64, &mut actual_instructions)?;
```

Comment:
Could replacing this with `read_8`  make the code simpler?
I know the code works as it is, but I'm afraid this will be criticized once we want to merge the branch into master...

Thank you

Reply:
Definetly, updated!